OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -135,6 +135,10 @@ public bool InterpretCommand(S server, Connection conn, Session.Client client, s
 
 						client.Slot = s;
 						S.SyncClientToPlayerReference(client, server.MapPlayers.Players[s]);
+
+						var validatedColor = ColorValidator.ValidatePlayerColorAndGetAlternative(server, client.Color, client.Index, conn);
+						client.PreferredColor = client.Color = validatedColor;
```

Comment:
I don't think you should be changing the PreferredColor?  The purpose of that field is to store the color that the player *would* use if it wasn't rejected by the validator.